osmdroid-shape/src/main/java/org/osmdroid/shape/ShapeConverter.java

```diff
@@ -39,7 +41,22 @@
 
 public class ShapeConverter {
 
-    public static List<Overlay> convert(MapView map, File file, ValidationPreferences prefs) throws Exception {
+    /**
+     *
+     * @param map the MapView to which these overlays will be added.
+     * @param file the shape file to be converted.
+     * @param prefs  allows the client to relax the level of validation when reading a shape file.
+     * @param fillColor fill color of polygon.
+     * @param outLineColor out line color of polyOverlay.
+     * @param width out line width of polyOverlay.
+     * @param getBubbleText customize titles, snippets, and sub-descriptions of bubbles.
+     *                      remember to handling bounding box if your shape file has both points and poly-overlays.
+     *                      please also mind that getSnippet is called before getTitle, and before getSubDescription.
```

Comment:
We shouldn't care about getSnippet being called before getTitle and getSubDescription: we can do that in `ShapeMetadataReader.read` at once in the right order.